modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTCollectionServiceImpl.java

```diff
@@ -190,6 +190,11 @@ public void publishCTCollection(long userId, long ctCollectionId)
 				"CTCollection is already published");
 		}
 
+		if (ctCollection.getStatus() != WorkflowConstants.STATUS_DRAFT) {
+			throw new CTCollectionStatusException(
+				"CTCollection is unable to be published");
+		}
+
```

Comment:
@david-truong does this seem ok to you? Is there any other status that we'd want to allow a publication to be published?

Reply:
it can also be pending

Reply:
Isn't the publication set to pending when it's being published? https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTProcessLocalServiceImpl.java#L76

We don't show the pending publications in the UI, so it probably doesn't matter. However, i think we should try to avoid multiple ctProcesses/backgroundTasks to be added while a publication is being published.

Reply:
What is the status when you schedule the publication? isn't it pending?

Reply:
scheduled publication has status SCHEDULED

Reply:
https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/internal/scheduler/PublishSchedulerImpl.java#L189